src/ape/utils.py

```diff
@@ -1,17 +1,15 @@
 import collections
```

Comment:
The changes to this file comprise of two things:

1. Delete a method related provider error handling. We have since learned providers should handle their errors on their own since they are always unique.
2. Add a couple new methods that get used elsewhere.